File: Coding_Questions/JavaScript/MaxDepthOfBinaryTree.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

const maxDepth = (root) =>
  !root ? 0 : Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;

File: Coding_Questions/Python/MaxDepthOfBinaryTree.py
# ---Max Depth of Binary Tree---

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def max_depth(self, root: TreeNode, depth: int = 0) -> int:
    return 0 if root is None else max(self.maxDepth(root.left),
                                      self.maxDepth(root.right)) + 1
